test/e2e/capi_test.go

```diff
@@ -116,4 +116,16 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
 			}
 		})
 	})
+
+	Context("Should successfully exercise machine pools", func() {
+		capi_e2e.MachinePoolSpec(context.TODO(), func() capi_e2e.MachinePoolInput {
```

Comment:
thoughts on adding on also adding an Azure version of a test with machine pool template that runs in PRs? Something like the ipv6 spec but with machine pools, so we can test for regression in the vmss service, proper LB provisioning, etc.

Reply:
Is this something you're trying to get done in this PR or is the plan to do it as a follow up?

Reply:
Just didn't have time. I'll add it this evening. Should be ready for review tomorrow.

Reply:
Added with LBSpec